### DISCUSSION TITLE
Identifying anti-Bredt Bridgehead compounds?

### DISCUSSION THREAD

**Discussion by paconius** [rdkit/rdkit#6157] [Q&A] [answered]
There are a few threads floating on the interwebs on identifying / counting bridgehead atoms. RDKit even has its own counter within rdMolDescriptors.

However, I haven't found a discussion about how to generally identify compounds that violate [Bredt's rule](https://en.wikipedia.org/wiki/Bredt%27s_rule), which states that a bridgehead atom cannot be in a double bond (or be aromatic). Sometimes these wacky compounds get generated from reaction transformations and it would be great to be able to filter them out.

Has anyone figured out a SMARTS pattern or other way to address this topic?

**Comment by greglandrum** [comment, marked as answer]
@paconius Based on the complexity of identifying bridghead atoms in general, I don't think you can do this with a SMARTS.
This would be pretty straightforward if the RDKit allowed you to get the bridgehead atoms (instead of just their counts) from Python, but that's unfortunately not currently possible (I will fix this for the next release). In the meantime you would have to duplicate the logic that's used to identify bridgeheads and work from that. The next release is coming in ~6 weeks, so I'll let you decide how urgent this is. :-)

**Reply by paconius** [reply]
Thanks for the repsonse, Greg. This is not a high priority issue for me at the moment. I can come back to this issue once the next version is released in ~6 weeks.  Thank you.

**Comment by dehaenw** [comment]
I have been using this function to check for bredt violations, which returns True if there is a violation:
```python
def bredt_violation(mol):
    """filter to check if there's a violation of bredts rule
    (no bridgehead double bonds). in this case also no aromatic bonds are
    allowed so it also filters cyclophane type compounds (intendedly)
    """
    problem=False
    sssr_idx = [set(x) for x in list(Chem.GetSymmSSSR(mol))]
    for i,ring1 in enumerate(sssr_idx):
        for j,ring2 in enumerate(sssr_idx):
            if i>j:
                intersect_idx=ring1.intersection(ring2)
                if len(intersect_idx)>2:
                    for idx in intersect_idx:
                        bh=False
                        neighbors = [a.GetIdx() for a
                                     in mol.GetAtomWithIdx(idx).GetNeighbors()]
                        for nidx in neighbors:
                            if nidx not in intersect_idx:
                                bh=True
                        if bh==True:
                            bondtypes = [b.GetBondType() for b in
                                         mol.GetAtomWithIdx(idx).GetBonds()]
                            if set(bondtypes) != {Chem.rdchem.BondType.SINGLE}:
                                problem=True
    return problem
```
i have not tested it thoroughly but it worked in my use case.

**Comment by rogersayle** [comment]
Interesting problem!  Fortunately, there appear to be only 16 ring systems (substructures) with S<7, i.e. that must obey Bredt's rule.
These can be rapidly tested by SMARTS matching in RDKit, where the first atom is a bridgehead atom that can't be conjugated.
```
*12*(*1)*2      bicyclo[0.1.1]butane (S=2)
*12*(*1)**2     bicyclo[0.1.2]pentane (S=3)
*12*(*1)***2    bicyclo[0.1.3]hexane (S=4)
*12*(*1)****2   bicyclo[0.1.4]heptane (S=5)
*12*(*1)*****2  bicyclo[0.1.5]octane (S=6)
*12*(**1)**2    bicyclo[0.2.2]hexane (S=4)
*12*(**1)***2   bicyclo[0.2.3]heptane (S=5)
*12*(**1)****2  bicyclo[0.2.4]octane (S=6)
*12*(***1)***2  bicyclo[0.3.3]octane (S=6)
*12**(*1)*2     bicyclo[1.1.1]pentane (S=3)
*12**(*1)**2    bicyclo[1.1.2]hexane (S=4)
*12**(*1)***2   bicyclo[1.1.3]heptane (S=5)
*12**(*1)****2  bicyclo[1.1.4]octane (S=6)
*12**(**1)**2   bicyclo[1.2.2]heptane (S=5)
*12**(**1)***2  bicyclo[1.2.3]octane (S=6)
*12***(**1)**2  bicyclo[2.2.2]octane (S=6)
```
Perhaps these can be added to RDKit's MolVS, as an additional sanity check.
Note that the fusion bond in naphthalene can be a double bond (S=8), so for a solution to be correct it needs to test ring sizes.

**Reply by dehaenw** [reply]
This is a really nice way to think about it. However, I would say your structures 1-9 are not bridgehead compounds in the sense used by organic chemists using Bredt's rule (i.e. for them the value between the brackets all need to be > 0). Though indeed due to small ring sizes many of these will not be able to accommodate a double bond on the fusion point, the 8 and 9 are stable structures. You can find structure 8 in Δ6-protoilludene, Tsugicoline A and structure 9 in cyperene. Nature approves! 

On the other hand, I think sometimes S some what higher than 6 will also be considered a red flag and Bredt forbidden, despite the nifty examples found in the literature. Even a bridgehead alkene like taxol where the bicycle is really large is considered at least exotic.

**Comment by RobertArbon** [comment]
Not super helpful but all of the Bredt violations have been coded up in C in https://github.com/StructureGenerator/surge

**Comment by papillot** [comment]
As a follow-up, I would like to suggest this SMARTS: `[R2,R3;$([x3]@[x2R2])]@&=[R]`

- The combination `[R2,R3;...x3...]` stands for an atom that belongs to 2 rings (or 3 rings) and with 3 ring connections. Belonging to 3 rings is an edge case for when the bridge has the same size as the other ring edges. In that case the SSSR size is 3, not 2.
- To further specify that it's a bridgehead, the context `@[x2R2]` refers to a connection to a bridge atom (belongs to 2 rings but only makes 2 ring connections). 
- `@&=` is to specify the offending double bond, also a ring bond. 
- `@` in the context is necessary in the case the offending double bond is part of the bridge.

I'd be glad to know if there are counterexamples?

**Reply by papillot** [reply]
Realized that `[R2,R3;...x3]` is the same as `[x3]` altogether because an atom that has 3 ring connections is de-facto part of 2 or 3 rings. So, an updated SMARTS could be: 

`[x3;$(*@[x2R2])]@!-[R]`

Where `@!-` also takes into account aromatic bonds as well as double bonds as suggested in @dehaenw function above
